packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -299,56 +320,113 @@ class BabelPluginAddPluginsToGatsbyConfig {
             )
 
             if (shouldAdd) {
-              const plugins = pluginNodes.value.elements.map(getPlugin)
-              const matches = plugins.filter(plugin => {
-                if (!key) {
-                  return plugin.name === pluginOrThemeName
-                }
-
-                return plugin.key === key
-              })
+              if (t.isCallExpression(pluginNodes.value)) {
```

Comment:
This could be almost anything tbh. If you're looking to explicitly support the `.filter(Boolean)` case then you can explicitly check for that, either step by step explicitly, or through a subtree match. I can't get this to work quickly but something like this used to work (but this explicit example does not):

```js
const same = t.isCallExpression(path.node, {
  callee: t.memberExpression(
    path.node.callee.object, 
    t.identifier('filter'),
  ),
  arguments: [t.identifier('Boolean')],
});
```

Good luck getting docs on that. There's also a template matching function. Might be too slow or janky for this purpose, might also be fine.

Reply:
Another use case is `.concat()` so not too keen on explicitly checking it.